### PR TITLE
Feature/oauth

### PR DESCRIPTION
**Pull request by DnOberon**
## Description
The following user stories are completed with this PR

A User can log in and be shown the Profile page
A redirected User can log in and be shown the Authorization page, or redirected immediately
A redirected User can register and be shown the Authorization page immediately upon completion
A User can successfully validate their email - and navigate back to the Profile page
A User can request a password reset, and receive an email with a redirect link
A User can reset their password after following an emailed link
A User can view their API keys
A User can delete any of their API keys
A User can generate a new API key
A User can view their OAuth Applications
A User can delete any of their OAuth Applications
A User can create a new OAuth Application
A redirected User can Authorize an OAuth application
A User can logout and be shown the login page
A User should be able to see details about the application requesting authorization on the Authorize page

## Motivation and Context
Deep Lynx needed to morph into a full fledged identity service in order to handle multiple outside application integrations. 

## How Has This Been Tested?
Multiple test suite test written. Integration tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/api/oauth_routes.ts [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+                    _error: req.query.error,
+                    api_key: req.query.newKey,
+                    api_secret: req.query.newSecret
+                })
+                return
+            })
+            .catch((err) => res.render('profile', {_error: err}))
+    }
+
+    private static createOAuthApplicationPage(req: Request, res: Response) {
```
Comment (jerenmb) [inline src/api/oauth_routes.ts:74]: Maybe consider renaming this method so that it's clear nothing is being created?

Comment (chrisritter) [reply src/api/oauth_routes.ts:74]: text_ops column is missing :)

Comment (chrisritter) [reply src/api/oauth_routes.ts:74]: Good to Merge once @jerenmb is addressed and text_ops is fixed

**Review by jerenmb — APPROVED** [review]

File: src/api/oauth_routes.ts (hunk 1 above)
Comment (DnOberon) [reply src/api/oauth_routes.ts:89]: Both addressed

**Pull request merged**